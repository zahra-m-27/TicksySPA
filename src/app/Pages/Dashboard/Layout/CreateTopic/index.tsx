import React from "react";
import { Button } from "antd";
import Assets from "../../../../Assets";
import styles from "./styles.module.scss";
import SEInput from "../../../../Components/SEInput";
import Picture from "../../../../Assets/Svgs/components/camera.svg";

export default function CreateTopic() {
  return (
    <div className={styles.container}>
      <div className={styles.rectangle}>
        <div className={styles.header}>
          تاپيک جدید
          <img src={Assets.Svgs.Topic} />
        </div>
        <div className={styles.upload}>
          <label htmlFor="picture">
            <img src={Picture} className={styles.picture} />
          </label>
          <input type="file" id="picture" className={styles.upload_image} />
        </div>
        <div className={styles.middle}>
          <label>:عنوان</label>
          <SEInput
            onChangeText={() => {}}
            innerContainerClassName={styles.input}
          />
          <label>:شناسه</label>
          <SEInput
            onChangeText={() => {}}
            innerContainerClassName={styles.input}
          />
        </div>
        <div className={styles.below}>
          <label>:توضيح</label>
          <SEInput
            minLines={5}
            onChangeText={() => {}}
            innerContainerClassName={styles.input}
          />
        </div>
        <Button type="primary" className={styles.enter_button}>
          ثبت
        </Button>
      </div>
    </div>
  );
}
