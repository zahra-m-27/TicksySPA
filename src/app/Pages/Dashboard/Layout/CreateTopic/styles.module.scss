$header-background-color: #265ecd;
.container {
  height: 100%;
}

.rectangle {
  box-shadow: 1px 2px 2px 1px #707070;
  width: 390px;
  height: 450px;
  margin-top: 3%;
  object-fit: contain;
  margin-left: 31%;
  border-radius: 10% 3%;
}

.header {
  width: 390px;
  height: 40px;
  background-color: $header-background-color;
  opacity: .9;
  display: flex;
  color: #fff;
  font-size: 13px;
  border-radius: 29% 12% 0 0;
  padding-right: 4%;
  align-items: center;
  justify-content: flex-end;

  img:nth-child(1) {
    width: 25px;
    height: 24px;
    margin: 0 0 0 2vw;
    color: #fff;
    border-color: #fff;
  }
}

.upload {
  display: flex;
  margin-top: 5px;
  align-items: center;
  flex-direction: column;
  width: clamp(400px, 30%, 566px);
}

.picture {
  object-fit: none;
  padding-top:2px;
  border-radius: 50%;
  border: solid 3px #265ecd;
  width: clamp(60px, 7vw, 100px);
  height: clamp(60px, 7vw, 100px);


}

.upload_image {
  display: none;
}

.middle {
  width: 70%;
  margin-right: 25%;
  text-align: right;
  padding-bottom: 5px;
}

.middle label {
  display: block;
  margin: 5px;
  text-align: right;
  font-size: 12px;
  color: #707070;
  opacity: 0.8;
  direction: rtl;
}

.middle input {
  width: 80%;
  height: 35px;
  border-radius: 5px;
  padding: 5px;
  direction: rtl;
  box-shadow: .5px .5px 1px 1px #707070;
  border: none;
}

@media(min-width: 500px) {
  .middle {
    width: 450px;
  }
}

.below {
  width: 70%;
  margin-right: 25%;
  text-align: right;
  padding-bottom: 15px;
}

.below label {
  display: block;
  margin: 5px;
  text-align: right;
  font-size: 12px;
  color: #707070;
  opacity: 0.8;
  direction: rtl;
}

.below input {
  width: 80%;
  height: 85px;
  border-radius: 5px;
  direction: rtl;
  box-shadow: .5px .5px 1px 1px #707070;
  border: none;

}

@media(min-width: 500px) {
  .below {
    width: 450px;
  }
}

.enter_button {
  background-color: $header-background-color;

  width: 40%;
  height: 32px;
  align-self: center;
  opacity: .9;
}
