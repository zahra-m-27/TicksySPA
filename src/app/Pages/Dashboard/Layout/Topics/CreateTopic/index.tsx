import API from '../../../../../API';
import {Button, message} from 'antd';
import Assets from '../../../../../Assets';
import {useRef, useState} from 'react';
import styles from './styles.module.scss';
import {useHistory} from 'react-router-dom';
import SEInput from '../../../../../Components/SEInput';

export default function CreateTopic() {
  const history = useHistory();
  const [Title, setTitle] = useState('');
  const [Avatar, setAvatar] = useState<File>();
  const [Loading, setLoading] = useState(false);
  const [AvatarUrl, setAvatarUrl] = useState<any>();
  const usernameRef = useRef<HTMLInputElement>(null);
  const [Description, setDescription] = useState('');
  const descriptionRef = useRef<HTMLTextAreaElement>(null);
  const [TitleHasError, setTitleHasError] = useState(false);
  const [DescriptionHasError, setDescriptionHasError] = useState(false);

  const onTitleEnter = () => {
    if (usernameRef && usernameRef.current) {
      usernameRef.current.focus();
    }
  };
  const onUsernameEnter = () => {
    if (descriptionRef && descriptionRef.current) {
      descriptionRef.current.focus();
    }
  };

  const onCreateTopic = () => {
    if (!Title) {
      setTitleHasError(true);
    }
    if (!Description) {
      setDescriptionHasError(true);
    }
    if (!Avatar) {
      message.error('برای تاپیک خود یک آواتار انتخاب کنید');
    }
    if (!Title || !Description || !Avatar) {
      return;
    }
    setTitleHasError(false);
    setDescriptionHasError(false);

    setLoading(true);
    API.Topics.CreateTopic({
      title: Title,
      avatar: Avatar,
      description: Description,
    })
      .then(() => {
        message.success('تاپیک با موفقیت ایجاد شد.');
        history.push('/dashboard/topics');
      })
      .catch((error) => {
        if (error.status === 403) {
          message.error('برای این عملیات نیاز هست تا احراز هویت کرده باشید');
        } else {
          message.error('انجام این عملیات ممکن نیست');
        }
      })
      .finally(() => setLoading(false));
  };

  return (
    <div className={styles.container}>
      <div className={styles.rectangle}>
        <div className={styles.header}>
          تاپيک جدید
          <img src={Assets.SVGs.Topic} alt="" />
        </div>
        <div className={styles.upload}>
          <label htmlFor="picture">
            <img
              alt=""
              className={styles.picture}
              src={AvatarUrl ?? Assets.Images.GoogleImage2}
            />
          </label>
          <input
            type="file"
            id="picture"
            className={styles.upload_image}
            onChange={(e) => {
              if (e.target.files) {
                setAvatar(e.target.files[0]);
                const fr = new FileReader();
                fr.onload = function () {
                  setAvatarUrl(fr.result);
                };
                fr.readAsDataURL(e.target.files[0]);
              }
            }}
          />
        </div>
        <div className={styles.middle}>
          <label>:عنوان</label>
          <SEInput
            content={Title}
            onEnter={onTitleEnter}
            onChangeText={setTitle}
            hasError={TitleHasError}
            innerContainerClassName={styles.input}
          />
        </div>
        <div className={styles.below}>
          <label>:توضيح</label>
          <SEInput
            minLines={5}
            ref={descriptionRef}
            content={Description}
            onChangeText={setDescription}
            hasError={DescriptionHasError}
            innerContainerClassName={styles.input}
          />
        </div>
        <Button
          loading={Loading}
          type="primary"
          className={styles.enter_button}
          onClick={onCreateTopic}>
          ثبت
        </Button>
      </div>
    </div>
  );
}
