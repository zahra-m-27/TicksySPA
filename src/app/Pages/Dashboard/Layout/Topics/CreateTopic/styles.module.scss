$header-background-color: #265ecd;
.container {
}

.rectangle {
  display: flex;
  margin: 3% auto 0;
  object-fit: contain;
  border-radius: 10% 3%;
  flex-direction: column;
  width: clamp(390px, 28vw, 523px);
  box-shadow: 1px 2px 2px 1px #7070709c;
}

.header {
  opacity: 0.9;
  color: #fff;
  width: 100%;
  height: 61px;
  display: flex;
  font-size: 16px;
  padding-right: 4%;
  align-items: center;
  justify-content: flex-end;
  border-radius: 50px 10px 0 0;
  background-color: $header-background-color;

  img:nth-child(1) {
    width: 25px;
    color: #fff;
    height: 24px;
    margin: 0 0 0 15px;
    border-color: #fff;
  }
}

.upload {
  display: flex;
  margin: 25px 0;
  align-items: center;
  flex-direction: column;
}

.picture {
  border-radius: 50%;
  border: solid 3px #265ecd;
  width: clamp(60px, 7vw, 100px);
  height: clamp(60px, 7vw, 100px);
}

.upload_image {
  display: none;
}

.middle {
  width: 100%;
  text-align: right;
  padding: 0 2rem 15px;
}

.middle label {
  display: block;
  margin: 5px;
  text-align: right;
  font-size: 12px;
  color: #707070;
  opacity: 0.8;
}

.input {
  box-shadow: unset !important;
}

.middle input {
  width: 100%;
  height: 35px;
  border: none;
  padding: 5px;
  border-radius: 5px;
  box-shadow: 0.5px 0.5px 1px 1px #707070;
}

.below {
  width: 100%;
  text-align: right;
  padding: 0 2rem 15px;
}

.below label {
  display: block;
  margin: 5px;
  text-align: right;
  font-size: 12px;
  color: #707070;
  opacity: 0.8;
}

.below input {
  width: 100%;
  height: 85px;
  border-radius: 5px;
  box-shadow: 0.5px 0.5px 1px 1px #707070;
  border: none;
}

.enter_button {
  width: 40%;
  opacity: 0.9;
  height: 32px;
  align-self: center;
  margin-bottom: 15px;
  background-color: $header-background-color;
}
