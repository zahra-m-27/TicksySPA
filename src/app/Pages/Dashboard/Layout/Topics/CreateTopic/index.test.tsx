import ReactDOM from 'react-dom';
import React from 'react';
import {render} from '@testing-library/react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';
import CreateTopic from './index';

const sample = (
  <MemoryRouter>
    <CreateTopic />
  </MemoryRouter>
);

test('render in ReactDOM', () => {
  const div = document.createElement('div');
  ReactDOM.render(sample, div);
});

it('render header', () => {
  const {getByText} = render(sample);
  getByText(/تاپيک جدید/i);
});

it('render subject', () => {
  const {getByText} = render(sample);
  getByText(/عنوان/i);
});

it('render description', () => {
  const {getByText} = render(sample);
  getByText(/توضيح/i);
});

it('render submit', () => {
  const {getByText} = render(sample);
  getByText(/ثبت/i);
});

it('snapshot', () => {
  const snapshot = renderer.create(sample).toJSON();
  expect(snapshot).toMatchSnapshot();
});
