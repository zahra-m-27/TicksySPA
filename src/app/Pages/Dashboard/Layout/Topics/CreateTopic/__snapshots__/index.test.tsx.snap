// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="container"
>
  <div
    className="rectangle"
  >
    <div
      className="header"
    >
      تاپيک جدید
      <img
        alt=""
        src="topic.svg"
      />
    </div>
    <div
      className="upload"
    >
      <label
        htmlFor="picture"
      >
        <img
          alt=""
          className="picture"
          src="google_images_2.png"
        />
      </label>
      <input
        className="upload_image"
        data-testid="avatar"
        id="picture"
        onChange={[Function]}
        type="file"
      />
    </div>
    <div
      className="middle"
    >
      <label>
        :عنوان
      </label>
      <div
        className="input_container"
        data-testid="input-container"
      >
        <div
          className="inner_container input"
        >
          <div
            className="content"
          >
            <input
              className="input"
              data-testid="title"
              dir="rtl"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder=""
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="below"
    >
      <label>
        :توضيح
      </label>
      <div
        className="input_container"
        data-testid="input-container"
      >
        <div
          className="inner_container input"
        >
          <div
            className="content"
          >
            <textarea
              className="input"
              data-testid="description"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder=""
              rows={5}
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <button
      className="ant-btn ant-btn-primary enter_button"
      data-testid="submit"
      onClick={[Function]}
      type="button"
    >
      <span>
        ثبت
      </span>
    </button>
  </div>
</div>
`;
